pkg/convert/convert_unit_test.go

```diff
@@ -0,0 +1,312 @@
+		fmt.Println("Error:", err)
+	}
+
+	purls := []string{}
+
+	if _, exists := data["spdxVersion"]; exists {
+		packages := data["packages"].([]interface{})
+		for _, pkg := range packages {
+			packageMap := pkg.(map[string]interface{})
+			if externalRefs, ok := packageMap["externalRefs"].([]interface{}); ok {
+				for _, ref := range externalRefs {
+					refMap := ref.(map[string]interface{})
+					if refType, ok := refMap["referenceType"].(string); ok && refType == "purl" {
+						if refLocator, ok := refMap["referenceLocator"].(string); ok {
+							purls = append(purls, refLocator)
+						}
+					}
+				}
+			}
+		}
+	} else {
+		components := data["components"].([]interface{})
+		for _, component := range components {
+			componentMap := component.(map[string]interface{})
+			if purl, ok := componentMap["purl"].(string); ok {
+				purls = append(purls, purl)
+			}
+
+			if subcomponents, ok := componentMap["components"].([]interface{}); ok {
+				for _, subcomponent := range subcomponents {
+					subcomponentMap := subcomponent.(map[string]interface{})
+					if purl, ok := subcomponentMap["purl"].(string); ok {
+						purls = append(purls, purl)
+					}
+				}
+			}
+		}
+	}
+	return len(purls)
+}
+
+func GetLicensesCount(json_str string) int {
+	var data map[string]interface{}
+	err := json.Unmarshal([]byte(json_str), &data)
+	if err != nil {
+		fmt.Println("Error:", err)
+	}
+
+	licenses := map[string]struct{}{}
+
+	if _, exists := data["spdxVersion"]; exists {
+		packages, _ := data["packages"].([]interface{})
+		for _, pkg := range packages {
+			packageMap, _ := pkg.(map[string]interface{})
+
+			if licenseConcluded, ok := packageMap["licenseConcluded"].(string); ok {
+				concludedLicenses := ExtractLicenses(licenseConcluded)
+				for _, lic := range FilterLicenses(concludedLicenses) {
+					licenses[lic] = struct{}{}
+				}
+			}
+
+			// if licenseDeclared, ok := packageMap["licenseDeclared"].(string); ok {
+			// 	declaredLicenses := ExtractLicenses(licenseDeclared)
+			// 	for _, lic := range FilterLicenses(declaredLicenses) {
+			// 		licenses[lic] = struct{}{}
+			// 	}
+			// }
+		}
+	} else {
+		components, _ := data["components"].([]interface{})
+		for _, component := range components {
+			componentMap, _ := component.(map[string]interface{})
+			if licensesList, ok := componentMap["licenses"].([]interface{}); ok {
+				for _, license := range licensesList {
+					if licenseMap, ok := license.(map[string]interface{}); ok {
+						if licenseIDMap, ok := licenseMap["license"].(map[string]interface{}); ok {
+							if licenseID, ok := licenseIDMap["id"].(string); ok {
+								licenses[licenseID] = struct{}{}
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+	return len(licenses)
+}
+
+func DownloadSBOMs() {
+	fileURL := "https://drive.usercontent.google.com/download?id=1LgGlq3g_H02mhzkc94cUd0zzxy0JhFim&export=download&authuser=0&confirm=t&uuid=483eac07-f1af-4356-abeb-4ba254e32b86&at=APZUnTWjSNLUgCQ8wwFZjsLS7Y36:1694113089657"
```

Comment:
This is scary. Who controls this content? Can it be changed without the end user's knowledge? I don't see a checksum, etc for us to use to know the target is what we expect?